lib/spack/docs/packaging_guide.rst

```diff
@@ -728,6 +728,102 @@ versions. See the documentation on `attribute_list_url`_ and
     syntax errors, or the ``import`` will fail.  Use this once you've
     got your package in working order.
 
+--------------------
+Finding new versions
+--------------------
```

Comment:
I moved this section up to be directly below "Versions and fetching" instead of down below "Handling RPATHs". I think it makes much more sense for all of the version/URL stuff to be together.